src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -82247,6 +82235,69 @@ static void F(object o)
             comp.VerifyTypes();
         }
 
+        /// <summary>
+        /// Nullability of variable members is tracked up to a fixed depth.
+        /// </summary>
+        [Fact]
+        [WorkItem(31395, "https://github.com/dotnet/roslyn/issues/31395")]
+        public void InheritNullabilityMaxDepth_01()
+        {
+            var source =
+@"#pragma warning disable 8618
+class A
+{
+    internal B? B;
+}
+class B
+{
+    internal A? A;
+}
+class Program
+{
+    static void F()
+    {
+        var a1 = new A() { B = new B() };
+        var a2 = new A() { B = new B() { A = a1 } };
+        var a3 = new A() { B = new B() { A = a2 } };
+        a1.B.ToString();
+        a2.B.A.B.ToString();
+        a3.B.A.B.A.B.ToString();
```

Comment:
Consider putting a `// 1` on the warning lines here and in the below test.

Reply:
Will add these comments in the next PR.

---
In reply to: [284816607](https://github.com/dotnet/roslyn/pull/35720#discussion_r284816607) [](ancestors = 284816607)